salt/modules/cmdmod.py

```diff
@@ -245,6 +271,8 @@ def _run(cmd,
     '''
     Do the DRY thing and only call subprocess.Popen() once
     '''
+    if 'pillar' in kwargs and not pillar_override:
+        pillar_override = kwargs['pillar']
```

Comment:
@cachedout if this works I think it is a lot cleaner, since we don't otherwise accept a kwarg called pillar. Do you see any issues here?